cni-plugin/main.go

```diff
@@ -260,8 +280,9 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return types.PrintResult(conf.PrevResult, conf.CNIVersion)
 	}
 
-	logrus.Debug("linkerd-cni: no previous result to pass through, emptying stdout")
-	return nil
+	logrus.Debug("linkerd-cni: no previous result to pass through, assume stand-alone run, send ok")
+
+	return types.PrintResult(&cniv1.Result{CNIVersion: cniv1.ImplementedSpecVersion}, conf.CNIVersion)
```

Comment:
As far as I remember, if Multus does not get something "meaningful" (which means CNIVersion at least), it fails to add networks to a Pod and the Pod's creation fails with error similar to:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_linkerd-destination-696d48756d-p4wxs_linkerd_cfbd3a42-c6f1-4eb4-9cc1-b57a0bc95b8d_0(2661b3cd99b38bd18ca498929cf2be89db85a3b1f2e33e859fb63910ea1058a1): error adding pod linkerd_linkerd-destination-696d48756d-p4wxs to CNI network "multus-cni-network": plugin type="multus" name="multus-cni-network" failed (add): [linkerd/linkerd-destination-696d48756d-p4wxs/cfbd3a42-c6f1-4eb4-9cc1-b57a0bc95b8d:linkerd-cni]: error adding container to network "linkerd-cni": unexpected end of JSON input
```

This happens when Linkerd CNI is called by Multus which does not pass anything from previous CNI plugins (in my case it was Openshift 4.10 and OpenshiftOVN), so Linkerd CNI returns `nil` and logs `linkerd-cni: no previous result to pass through, emptying stdout`. I added this change in Linkerd CNI to make it return `something`.

Reply:
As for installing other CNI plugins, as far as I remember in case of Kubernetes clusters which have Multus, Multus changes previous cluster's CNI configuration and by default only processes the "first" CNI plugin as its `delegate` (as least if we install Multus as DaemonSet). This "first/default" plugin is supposed to be a normal network plugin such as Calico, Flannel, Kube-OVN etc.

Any other installed CNI plugins are silently ignored by Multus (as well as .conflist files) and must be attached via NetworkAttachmentDefinition.

As least it is how I understand the Multus quickstart (https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/quickstart.md)